source/extensions/filters/http/wasm/wasm_filter.h

```diff
@@ -33,19 +33,23 @@ class FilterConfig : Logger::Loggable<Logger::Id::wasm> {
       wasm = handle->wasmHandle()->wasm().get();
     }
     if (!wasm || wasm->isFailed()) {
-      if (plugin_->fail_open_) {
+      if (handle->plugin()->fail_open_) {
         // Fail open skips adding this filter to callbacks.
         return nullptr;
       } else {
         // Fail closed is handled by an empty Context.
-        return std::make_shared<Context>(nullptr, 0, handle);
+        return std::make_shared<Context>(
+            nullptr, 0, handle, envoy::config::core::v3::TrafficDirection::UNSPECIFIED, nullptr);
       }
     }
-    return std::make_shared<Context>(wasm, handle->rootContextId(), handle);
+    return std::make_shared<Context>(wasm, handle->rootContextId(), handle, direction_,
+                                     listener_metadata_);
   }
 
 private:
-  PluginSharedPtr plugin_;
+  envoy::config::core::v3::TrafficDirection direction_;
+  const envoy::config::core::v3::Metadata* listener_metadata_;
```

Comment:
both const? Here as http filter and below as network filter.

Reply:
done. Thanks! https://github.com/envoyproxy/envoy/pull/17243/commits/00a45f4ecde4e7cdca3173325b330c219b99be89